src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3808 @@
+#############################################
+function to_julia_matrix(A::Union{MatElem})
+  return eltype(A)[A[i, j] for i = 1:nrows(A), j = 1:ncols(A)]
+end
+
+function Base.reshape(M::MatElem, n, m)
+  julia_matrix = to_julia_matrix(M)
+  julia_matrix = reshape(julia_matrix, n, m)
+  R = base_ring(M)
+  mat_space = MatrixSpace(R, n, m)
+  return mat_space(julia_matrix)
+end
+
+function hom2(f1::MatElem{T}, g1::MatElem{T}) where T
+  R = base_ring(f1)
+  s1, s0 = size(f1)
+  t1, t0 = size(g1)
+
+  g2 = matrix_kernel(g1)
+  n = s1*t0
+  m = s0*t0 + s1*t1
+  delta::MatrixElem{T} = zero_matrix(R, m,n)
+  #delta::Matrix{T} = zeros(R, n, m)
+  for j=1:m
+    #b_vector::Vector{T} = [i == j ? R(1) : R(0) for i=1:m]
+    b_vector::MatrixElem{T} = zero_matrix(R, 1,m)
+    b_vector[1,j] = R(1)
+    #println(typeof(Core.Array(b_vector[1,1:s0*t0])))
+    A = reshape(b_vector[1,1:s0*t0], s0, t0)
+    #Mat = AbstractAlgebra.MatrixSpace(R, t0, s0)
+    #A = Mat(A)
+    B = reshape(b_vector[1,s0*t0+1:length(b_vector)], s1, t1)
+    #Mat = AbstractAlgebra.MatrixSpace(R, t1, s1)
+    #B = Mat(B)
+    res = f1*A - B*g1
+    res = reshape(res, 1, length(res))
+    for i=1:length(res)
+      delta[j,i] = res[1,i]
+    end
+  end
+
+  #Mat = AbstractAlgebra.MatrixSpace(R, size(delta)...)
+  #display(delta)
+  gamma = matrix_kernel(delta)
+
+  t2 = size(g2)[1]
+  n = m
+  m = s0*t1 + s1*t2
+  #rho::Matrix{T} = zeros(R, n, m)
+  rho::MatrixElem{T} = zero_matrix(R, m,n)
+  for j=1:m
+    #b_vector::Vector{T} = [R(0) for i=1:m]
+    b_vector = zero_matrix(R, 1,m)
+    b_vector[1,j] = R(1)
+    C = reshape(b_vector[1,1:s0*t1], s0, t1)
+    #Mat = AbstractAlgebra.MatrixSpace(R, t1, s0)
+    #C = Mat(C)
+    D = reshape(b_vector[1,s0*t1+1:length(b_vector)], s1, t2)
+    #Mat = AbstractAlgebra.MatrixSpace(R, t2, s1)
+    #D = Mat(D)
+    res1 = C*g1
+    res1 = reshape(res1, 1,length(res1))
+    res2 = f1*C - D*g2
+    res2 = reshape(res2, 1,length(res2))
+    for i=1:length(res1)
+        rho[j,i] = res1[1,i]
+    end
+    for i=1:length(res2)
+        #rho[:,j] = vcat(vec(res1), vec(res2))
+        rho[j,i+length(res1)] = res2[1,i]
+    end
+  end
+
+  M = SubQuo(gamma, rho)
+
+  function convert_to_matrix(v::SubQuoElem{T})
+    if parent(v) !== M
+      throw(DomainError("v does not represent a homomorphism"))
+    end
+    R = base_ring(M)
+    #A = AbstractAlgebra.matrix(R,map(x -> convert_poly(R,x),Array(v.singular_v)))
+    #A = map(x -> convert_poly(R,x),Array(v.singular_v))
+    #A = AbstractAlgebra.matrix(R, t0, s0, A[1:s0*t0])'
+    A = reshape(dense_row(v.repres.coords[R, 1:s0*t0], s0*t0), s0, t0)
+    #A = matrix(R, t0, s0, dense_row(v.repres.coords[R, 1:s0*t0], s0*t0))'
+    #A = v.v*v.parent_sq.generators  # ineffizienter
+    #A = reshape(A[1,1:s0*t0], s0, t0)
+    #N = cokernel(g1)
+    #for i=1:s0ems) <: Tuple ? [i for i in elems] : [elems])
+
+    #    row = lift(N,A[i,:])
+    #    for j=1:t0
+    #        A[i,j]=row[1,j]
+    #    end
+    #end
```

Comment:
looooots of commented out code.. why? Is this simply incomplete, and you plan to activate this code at a later time? Or is it a leftover from an experiment?